### DISCUSSION TITLE
CSS for SPA landing page

### DISCUSSION THREAD

**Discussion by mitar** [tailwindlabs/tailwindcss#10073] [Help]
I have Vue SPA app (using Vite) and I am using Tailwind CSS for all my CSS needs. As you know, SPAs (when server-side rendering is not enabled) first serve to the client a static HTML page which loads CSS and JS which then renders the app. Now, I would like to show a simple "loading" message to be shown to the user while CSS and JS is loading and before the app is rendered. I have something like:

```html
<!DOCTYPE html>
<html lang="en" class="text-gray-900 bg-slate-100 font-sans">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>My App</title>
    <script type="module" src="/src/main.ts"></script>
  </head>
  <body class="flex flex-col min-h-screen">
    <header></header>
    <main class="flex-grow flex justify-center">
      <div class="self-center jsonly">Loading...</div>
      <noscript class="self-center"><style>.jsonly{display: none}</style>JavaScript is required.</noscript>
    </main>
    <footer></footer>
  </body>
</html>
```

The issue is that those classes are not defined. So I would like to generate simplest Tailwind CSS styles for those and only those classes (without reset CSS) so that I could hard-code them into `<style></style>` tag inside this static file. My thinking is that I generate styles and then with a simple script try to replace existing style tag with those generated contents, instead of hard-coding them once and keeping them in-sync manually.

For example, manually I created (which works well for me):

```html
    <style>
      body{margin: 0}
      .text-gray-900{color: rgb(17 24 39)}
      .bg-slate-100{background-color: rgb(241 245 249)}
      .font-sans{font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"}
      .flex{display: flex}
      .flex-col{flex-direction: column}
      .flex-grow{flex-grow: 1}
      .min-h-screen{min-height: 100vh}
      .justify-center{justify-content: center}
      .self-center{align-self: center}
    </style>
```

Any suggestion how could I do that? I do not want to destroy my existing (regular) Tailwind CSS configuration.

**Comment by wongjn** [comment]
You could use the [PostCSS API](https://postcss.org/api/) directly to run Tailwind and get the output CSS to then inline. You would also need to [disable the core preflight plugin](https://tailwindcss.com/docs/preflight#disabling-preflight) and also enable the `experimental.optimizeUniversalDefaults` to reduce the footprint further, so perhaps use a different Tailwind configuration file via the [`@config` directive](https://tailwindcss.com/docs/functions-and-directives#config) that has your main Tailwind configuration as a preset.

**Reply by mitar** [reply]
Thanks. Do you know of any example of running Tailwind through PostCSS API I could look at? Especially because I am currently running everything through Vite so I worry there might be some extra steps involved?
